app/client/components/GridView.js

```diff
@@ -902,6 +904,24 @@ GridView.prototype.insertColumn = async function(colId = null, options = {}) {
   return newColInfo;
 };
 
+GridView.prototype.makeHeadersFromRow = async function(selection) {
+  if (this._getCellContextMenuOptions().disableMakeHeadersFromRow){
+    return;
+  }
+  const actions = this.viewSection.columns.peek().reduce((acc, col) => {
+    const colId = col.colId.peek();
+    let newColLabel = this.tableModel.tableData.getValue(selection.rowIds[0], colId);
+    if (Array.isArray(newColLabel) && newColLabel[0] === GristObjCode.List) {
```

Comment:
References are another type that will need special handling here. (Currently, the numeric id is what's used as the label.)